### PR TITLE
fix(curriculum): replace self-closing with void element#55342

### LINKED ISSUE #55342: Replace all references to self-closing tags in HTML with void element
### Describe the Issue

Elements that do not accept content (child nodes) are void or empty elements.

> Self-closing tags (`<tag />`) do not exist in HTML.

https://developer.mozilla.org/en-US/docs/Glossary/Void_element

https://html.spec.whatwg.org/multipage/syntax.html#void-elements

It doesn't matter if the void element has a trailing slash or not, it is still a void element, not a self-closing element.

Whether we want to promote trailing slashes on void elements is up for debate and a matter of taste/style. But we shouldn't call them self-closing elements.

---

For any challenge where the void element is explained, if they need to have the explanation updated, I suggest we reference MDN or the specs for help with the text.

MDN:

> A void element is an element in HTML that cannot have any child nodes (i.e., nested elements or text nodes). Void elements only have a start tag; end tags must not be specified for void elements.

Specs:

> [Void elements](https://html.spec.whatwg.org/multipage/syntax.html#void-elements) can't have any contents (since there's no end tag, no content can be put between the start tag and the end tag).

I would suggest we set aside one challenge step to thoroughly explain what a void element is and give a list of the void elements.

---

Search for self-closing in the new RWD curriculum returns these results:

- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-basic-css-by-building-a-cafe-menu/5f3ef6e01f288a026d709587.md #55430
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-basic-css-by-building-a-cafe-menu/5f46e8284aae155c83015dee.md #55430
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-basic-css-by-building-a-cafe-menu/5f331e55dfab7a896e53c3a1.md #55430
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-basic-css-by-building-a-cafe-menu/5f3477cb2e27333b1ab2b955.md #55430
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/6140c7e645d8e905819f1dd4.md #55445
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61695d1fbc003856628bf561.md #55457
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61696ef7ac756c829f9e4048.md #55457
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/616965351e74d4689eb6de30.md #55457
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-css-grid-by-building-a-magazine/614385513d91ae5c251c2052.md #55544
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-css-variables-by-building-a-city-skyline/5d822fd413a79914d39e98cc.md
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dc23f9bf86c76b9248c6eba.md #55383
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804d8.md #55383
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804dc.md #55383
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/62bb4009e3458a128ff57d5d.md #55383
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62cc5b1779e4d313466f73c5.md #55545
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-responsive-web-design-by-building-a-piano/612e83ec2eca1e370f830511.md #55547
- [x] https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f34ecc1091b4fd5a8a484.md #55611

---

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design

### Your code

```



```

### Expected behavior

Correct terminology usage.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

As an aside. I really do not think we have to keep calling the elements, void elements. Right now, we are repeating "self-closing" as if it needs to be mentioned every time.

### PR DESCRIPTION
**Pull request by uzma-nazim**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55342

### CODE REVIEW

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.md:19]:
```suggestion
Note that `img` elements are void elements.
```

or 

```suggestion
Note that `img` is a void element.
```

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-text-field.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-text-field.md:22]:
```suggestion
Note that `input` elements are void elements.
```

```suggestion
Note that `input` is a void element.
```

File: curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md:15]:
```suggestion
In HTML, almost all tags have both an opening and closing tag: `<div></div>`; the closing tag always has a forward slash before the tag name that you are closing. However, there are special instances in HTML called “void elements”, or tags that don’t require both an opening and closing tag before another tag can start.
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md:12]:
```suggestion
External CSS is the most common method you will come across, and it involves creating a separate file for the CSS and linking it inside of an HTML’s opening and closing `<head>` tags with a `<link>` element (which is a void element and doesn't require a closing tag):
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md:14]:
```suggestion
First, you add a `<link>` tag inside of the opening and closing `<head>` tags of the HTML file. The `href` attribute is the location of the CSS file, either an absolute URL or, what you’ll be utilizing, a URL relative to the location of the HTML file. In the example above, you are assuming both files are located in the same directory. The `rel` attribute is required, and it specifies the relationship between the HTML file and the linked file.
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-html-foundations/html-foundations-lesson-b.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/16-the-odin-project/top-learn-html-foundations/html-foundations-lesson-b.md:11]:
```suggestion
There are some HTML elements that do not have a closing tag. These elements often look like this: `<br />` or `<img/>`, but some can also be used without the closing forward slash such as `<br>` or `<img>`. These are known as void elements or empty elements because they don’t wrap any content. You will encounter a few of these in later lessons, but for the most part, elements will have both opening and closing tags.
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-html-foundations/html-foundations-lesson-b.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/16-the-odin-project/top-learn-html-foundations/html-foundations-lesson-b.md:21]:
```suggestion
Why are void elements like `<br/>` or `<img>` essential in HTML?
```

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-e.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-e.md:12]:
```suggestion
To display an image in HTML you use the `<img>` element. Unlike the other elements you have encountered, the `<img>` is a void element. Empty, void HTML elements do not need a closing tag.
```

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-f.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-f.md:11]:
```suggestion
To display an image in HTML you use the `<img>` element. Unlike the other elements you have encountered, the `<img>` is a void element. Empty, void HTML elements do not need a closing tag.
```

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-g.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-g.md:11]:
```suggestion
To display an image in HTML you use the `<img>` element. Unlike the other elements you have encountered, the `<img>` is a void element. Empty, void HTML elements do not need a closing tag.
```

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-h.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-h.md:11]:
```suggestion
To display an image in HTML you use the `<img>` element. Unlike the other elements you have encountered, the `<img>` is a void element. Empty, void HTML elements do not need a closing tag.
```

**Review by majestic-owl448 — COMMENTED** [review]
THis is a preliminary review, someone who is a native English speaker may have additional feedbacks.

**Review by uzma-nazim — COMMENTED** [review]
Applied review changes. Please check for any further suggestions!

**Comment by ilenia-magoni** [comment]
Please do not merge main into your branch. If needed a maintainer will ask, otherwise it just contributes to clog the tests queue.

**Comment by uzma-nazim** [comment]
> Please do not merge main into your branch. If needed a maintainer will ask, otherwise it just contributes to clog the tests queue.

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-placeholder-text-to-a-text-field.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-placeholder-text-to-a-text-field.md:20]:
```suggestion
**Note:** Remember that `input` is a void element.
```

File: curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md:15]:
```suggestion
In HTML, almost all tags have both an opening and closing tag: `<div></div>`; the closing tag always has a forward slash before the tag name that you are closing. However, there are special instances in HTML called “void elements”, which are elements that cannot have any content and whose syntax doesn't require both an opening and closing tag before another tag can start.
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md:12]:
```suggestion
External CSS is the most common method you will come across, and it involves creating a separate file for the CSS and linking it inside of an HTML’s opening and closing `<head>` tags with a `link` element. The `link` element is a void element, so should not have a closing tag.
```

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-e.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-e.md:12]:
```suggestion
To display an image in HTML you use the `<img>` element. Unlike the other elements you have encountered, the `<img>` is a void element. Void elements cannot have content and do not need a closing tag.
```

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-f.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-f.md:11]:
```suggestion
To display an image in HTML you use the `<img>` element. Unlike the other elements you have encountered, the `<img>` is a void element. Void elements cannot have content and do not need a closing tag.
```

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-g.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-g.md:11]:
```suggestion
To display an image in HTML you use the `<img>` element. Unlike the other elements you have encountered, the `<img>` is a void element. Void elements cannot have content and do not need a closing tag.
```

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-h.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-h.md:11]:
```suggestion
To display an image in HTML you use the `<img>` element. Unlike the other elements you have encountered, the `<img>` is a void element. Void elements cannot have content and do not need a closing tag.
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.md:19]: This change has not been applied.

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-placeholder-text-to-a-text-field.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-placeholder-text-to-a-text-field.md:20]: This change has not been applied.

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-text-field.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-text-field.md:22]: This change has not been applied.

File: curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md:15]: This change has not been applied.

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md:12]: This change has not been applied.

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations/css-foundations-lesson-f.md:14]: This change has not been applied.

File: curriculum/challenges/english/16-the-odin-project/top-learn-html-foundations/html-foundations-lesson-b.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/16-the-odin-project/top-learn-html-foundations/html-foundations-lesson-b.md:11]: This change has not been applied.

File: curriculum/challenges/english/16-the-odin-project/top-learn-html-foundations/html-foundations-lesson-b.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/16-the-odin-project/top-learn-html-foundations/html-foundations-lesson-b.md:21]: This change has not been applied.

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-e.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-e.md:12]: This change has not been applied.

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-f.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-f.md:11]: This change has not been applied.

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-g.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-g.md:11]: This change has not been applied.

File: curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-h.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/16-the-odin-project/top-links-and-images/links-and-images-lesson-h.md:11]: This change has not been applied.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Marking a conversation as resolved does not apply the suggestion.

**Comment by uzma-nazim** [comment]
> Marking a conversation as resolved does not apply the suggestion.

I applied the changes from the review but accidentally did a force push and deleted the commit history. I was facing issues while pushing the code, which is why all the resolved conversations were undone. Apologies for the inconvenience.

File: curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md:15]:
```suggestion
In HTML, almost all tags have both an opening and closing tag: `<div></div>`; the closing tag always has a forward slash before the tag name that you are closing. However, there are special instances in HTML called <dfn>void elements</dfn>, or tags that don’t require both an opening and closing tag before another tag can start.
```

Let's make this a proper definition.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/03-front-end-development-libraries/react/learn-about-self-closing-jsx-tags.md:15]:
```suggestion
In HTML, almost all tags have both an opening and closing tag: `<div></div>`; the closing tag always has a forward slash before the tag name that you are closing. However, there are special instances in HTML called <dfn>void elements</dfn>, or elements that don’t require both an opening and closing tag before another element can start.
```

**Review by naomi-lgbt — APPROVED** [review]

**Review by majestic-owl448 — APPROVED** [review]
Hi @uzma-nazim. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**
